test/jdk/java/awt/print/PrinterJob/SecurityDialogTest.java

```diff
@@ -21,210 +21,108 @@
  * questions.
  */
 
-/**
+import java.awt.Frame;
+import java.awt.print.PageFormat;
+import java.awt.print.PrinterJob;
+import java.lang.reflect.InvocationTargetException;
+
+import javax.print.PrintService;
+import javax.print.attribute.HashPrintRequestAttributeSet;
+import javax.print.attribute.PrintRequestAttributeSet;
+import javax.swing.BorderFactory;
+import javax.swing.Box;
+import javax.swing.JComponent;
+import javax.swing.JLabel;
+import javax.swing.SwingUtilities;
+
+/*
  * @test
  * @bug 4937672 5100706 6252456
  * @key printer
- * @run main/othervm/manual -Djava.security.manager=allow SecurityDialogTest
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @run main/manual/othervm -Djava.security.manager=allow SecurityDialogTest
  */
+public class SecurityDialogTest extends Frame {
+    private static final String INSTRUCTIONS =
+            "This test brings up a native and cross-platform page and print dialogs.\n" +
+            "\n" +
+            "If the dialog has an option to save to file, the option ought " +
+            "to be disabled.\n" +
+            "\n" +
+            "Press the Pass button if the \"Print to file\" option was disabled in\n" +
+            "all the dialogs where it was present.\n" +
+            "Otherwise, press the Fail button.\n" +
+            "\n" +
+            "The dialogs should be displayed even when " +
+            "there is no queuePrintJob permission.";
+
+    private static JLabel dialogType;
+
+    public static void main(String[] args) throws Exception {
+        if (PrinterJob.lookupPrintServices().length == 0) {
+            throw new RuntimeException("Printer not configured or available.");
+        }
 
-import java.awt.* ;
-import java.awt.print.* ;
-import java.io.*;
-import java.security.*;
-import javax.print.*;
-import javax.print.attribute.*;
-
-public class SecurityDialogTest {
-
-
-    public static void main ( String args[] ) {
-
-        String[] instructions =
-           {
-            "You must have a printer available to perform this test.",
-            "This test brings up a native and cross-platform page and",
-            "print dialogs.",
-            "The dialogs should be displayed even when ",
-            "there is no queuePrintJob permission.",
-            "If the dialog has an option to save to file, the option ought",
-            "to be disabled if there is no read/write file permission.",
-            "You should test this by trying different policy files."
-          };
+        PassFailJFrame passFailJFrame = new PassFailJFrame.Builder()
```

Comment:
I am getting an error for using new `PassFailJFrame passFailJFrame = new PassFailJFrame.Builder()`, stating _"'Builder()' has private access in 'PassFailJFrame.Builder'"_ Probably because I have fetched latest PassFailJFrame.
And also you can use `PassFailJFrame.builder()` method.

Reply:
For me its building and running properly

Reply:
Ok, might be an issue from my side. Will look into it.

Reply:
> I am getting an error for using `new PassFailJFrame passFailJFrame = new PassFailJFrame.Builder()`, stating *"'Builder()' has private access in 'PassFailJFrame.Builder'"*. Probably because I have fetched latest `PassFailJFrame`.

@TejeshR13 You're absolutely right. Thanks for catching it.

> And also you can use `PassFailJFrame.builder()` method.

Not *can* but *must*.

The test won't work after it's integrated, it won't compile.